test/functionalTest/cases/0988_put_with_param_type/0988_put_with_param_type.test

```diff
@@ -0,0 +1,257 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+PUT entity with type parameter in query
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. POST /v2/entities E1/T1
+# 02. POST /v2/entities E1/T2
+# 03. POST /v2/entities/E1 T1 (update attribute A1)
+# 04. GET /v2/entities (updated attribute A1 in E1/T1, not updated in E1/T2)
+# 05. POST/v2/entities/E1 T2 (append attribute A3)
+# 06. GET /v2/entities (added attribute A3 in E1/t2, not added in E1/T1)
+# 07. POST/v2/entities/E1 T2 (append attribute A3, error already exists)
+#
+
+
+echo "01. POST /v2/entities/E1 T1"
+echo "==========================="
+payload='{
+    "id": "E1",
+    "type": "T1",
+    "A1":{
+         "value": "V1 A1 E1/T1"
+    }
+}'
+orionCurl --url /v2/entities --payload "$payload" --json
+echo
+echo
+
+
+echo "02. POST /v2/entities/E1 T2"
+echo "==========================="
+payload='{
+    "id": "E1",
+    "type": "T2",
+    "A1":{
+         "value": "V1 A1 E1/T2"
+    },
+    "A2":{
+         "value": "V1 A2 E1/T2"
+    }
+}'
+orionCurl --url /v2/entities --payload "$payload" --json
```

Comment:
What happens if this payload (where type is set to T2) is sent with URI param type=T5 ?

Hint: 400 Bad Request (discussion in NGSIv2 needed to decide the behavior) ...

With the current implementation I believe the URI param silently overwrites the value in the payload.